pkg/controller/common/util.go

```diff
@@ -2030,6 +2032,16 @@ func CopyAllowedAnnotations(srcObj, dstObj metav1.Object) {
 	}
 }
 
+// CopyAllowedLabels copies allowed labels matching the validLabelsMatch regexp from the
+// source map to the destination object allowing overwrites
+func CopyAllowedLabels(srcLabels map[string]string, dstObj metav1.Object, overwrite bool) {
```

Comment:
sure you want to support "overwrite" here? seems to me like it's not really necessary but may be wrong

Reply:
IIUC we need overwrite to allow overriding labels on a DataSource when the env vars/labels of the containerdisk changed.

Reply:
Yeah I am saying just always overwrite to avoid the extra bool param. Unless that's a bad idea

Reply:
Ah I see, this func is shared between multiple places and some expect labels to not be overwritten.